### PR TITLE
[BE] Annotate subgraph_lowering

### PR DESCRIPTION
**Pull request by yanboliang**
Fixes #ISSUE_NUMBER

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

### CODE REVIEW

**Review by anijain2305 — APPROVED** [review]

File: torch/_inductor/subgraph_lowering.py [hunk 1]
```diff
@@ -85,18 +90,22 @@ def __init__(self, tracer, num_inputs):
...
+    def output(self, *args: Tuple[object]) -> torch.fx.Node:
         return self.tracer.create_node(
             "output", "output", (tuple(self.tracer.create_arg(a) for a in args),), {}
         )
 
 
-def lower_pointwise_subgraph(subgraph: ir.Subgraph, inputs: List[InputDescriptor]):
+def lower_pointwise_subgraph(
+    subgraph: ir.Subgraph, inputs: List[InputDescriptor]
+) -> Callable[[Any, Dict[str, Any]], Any]:
```
Comment (XuehaiPan) [inline torch/_inductor/subgraph_lowering.py:104]:
```suggestion
_P = ParamSpec("_P")

def lower_pointwise_subgraph(
    subgraph: ir.Subgraph, inputs: List[InputDescriptor]
) -> Callable[_P, Any]:
```

File: torch/_inductor/subgraph_lowering.py [hunk 2]
```diff
@@ -138,7 +147,7 @@ def fake_inner_fn(loop_idx, input_idx):
 
     lowered_gm = torch.fx.GraphModule({}, tracer.graph)
 
-    def inner_fn(*args, **kwargs):
+    def inner_fn(*args: Any, **kwargs: Dict[str, Any]) -> Any:
```
Comment (XuehaiPan) [inline torch/_inductor/subgraph_lowering.py:150]:
```suggestion
    def inner_fn(*args: _P.args, **kwargs: _P.kwargs) -> Any:
```

**Review by zou3519 — APPROVED** [review]
LGTM, but see XuehaiPan's comment

**Pull request closed without merging**